src/runtime/methodbinder.cs

```diff
@@ -161,11 +166,40 @@ internal MethodBase[] GetMethods()
                 // I'm sure this could be made more efficient.
                 list.Sort(new MethodSorter());
                 methods = (MethodBase[])list.ToArray(typeof(MethodBase));
+                pyArgumentConverter = this.GetArgumentConverter();
                 init = true;
             }
             return methods;
         }
 
+        IPyArgumentConverter GetArgumentConverter() {
+            IPyArgumentConverter converter = null;
+            Type converterType = null;
+            foreach (MethodBase method in this.methods)
+            {
+                var attribute = method.DeclaringType?
+                      .GetCustomAttributes(typeof(PyArgConverterAttribute), inherit: false)
+                      .OfType<PyArgConverterAttribute>()
+                      .SingleOrDefault()
+                    ?? method.DeclaringType?.Assembly
+                      .GetCustomAttributes(typeof(PyArgConverterAttribute), inherit: false)
+                      .OfType<PyArgConverterAttribute>()
+                      .SingleOrDefault();
```

Comment:
Did not want to overoptimize at first. Weren't there some issues with AppDomains for global caches?

Reply:
Oh yes, there is plenty of brokenness in that general direction, that's why I put the "(maybe per engine)" there. We have other global data though already, and for the one usecase of multiple AppDomains that I know of (reloading code in Unity) it works because they take down the PythonEngine and restart it, s.t. always exactly one is running. The comment was also more of a "fyi", not a requirement for merging :)